src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsVehicleRentalDataSource.java

```diff
@@ -31,214 +34,144 @@
+            bike.getLon() != null &&
+            bike.getLat() != null
+        ) {
             VehicleRentalStation brstation = new VehicleRentalStation();
-            brstation.id = new FeedScopedId(networkName, stationNode.path("station_id").asText());
-            brstation.vehiclesAvailable = stationNode.path("num_bikes_available").asInt();
-            brstation.spacesAvailable = stationNode.path("num_docks_available").asInt();
-            brstation.isKeepingVehicleRentalAtDestinationAllowed = config.allowKeepingRentedVehicleAtDestination();
+            brstation.id = new FeedScopedId(system, bike.getBikeId());
+            brstation.name = new NonLocalizedString(bike.getBikeId());
+            brstation.longitude = bike.getLon();
+            brstation.latitude = bike.getLat();
+            brstation.vehiclesAvailable = 1;
+            brstation.spacesAvailable = 0;
+            brstation.allowDropoff = false;
+            brstation.isFloatingBike = true;
             brstation.isCarStation = routeAsCar;
             return brstation;
+        } else {
+            return null;
         }
     }
 
-    // TODO This is not currently safe to use. See javadoc on GbfsVehicleRentalDataSource class.
-    class GbfsFloatingVehicleDataSource extends GenericJsonVehicleRentalDataSource<GbfsVehicleRentalDataSourceParameters> {
-
-        public GbfsFloatingVehicleDataSource(GbfsVehicleRentalDataSourceParameters config) {
-            super(config, "data/bikes");
-        }
-
-        @Override
-        public VehicleRentalStation makeStation(JsonNode stationNode) {
-            if (stationNode.path("station_id").asText().isBlank() &&
-                    stationNode.has("lon") &&
-                    stationNode.has("lat")
-            ) {
-                VehicleRentalStation brstation = new VehicleRentalStation();
-                brstation.id = new FeedScopedId(networkName, stationNode.path("bike_id").asText());
-                brstation.name = new NonLocalizedString(stationNode.path("name").asText());
-                brstation.longitude = stationNode.path("lon").asDouble();
-                brstation.latitude = stationNode.path("lat").asDouble();
-                brstation.vehiclesAvailable = 1;
-                brstation.spacesAvailable = 0;
-                brstation.allowDropoff = false;
-                brstation.isFloatingBike = true;
-                brstation.isCarStation = routeAsCar;
-                return brstation;
-            } else {
-                return null;
-            }
-        }
+    @Override
+    public String toString() {
+        return ToStringBuilder.of(GbfsVehicleRentalDataSource.class)
+                .addStr("url", url)
+                .addStr("language", language)
+                .addBoolIfTrue("routeAsCar", routeAsCar)
+                .addBoolIfTrue("allowKeepingRentedVehicleAtDestination", allowKeepingRentedVehicleAtDestination)
```

Comment:
FYI, When writing toStrings I try to put the most important fields first, and shorter fields before long one. I am not sure witch is the important field here, so this might be ok. Why? Because, in the debugger often only the first 50-80 characters is visible.

Reply:
The url is the only required and interesting field. The others are interesting only for debugging, and normally not needed or even displayed as they are null/false